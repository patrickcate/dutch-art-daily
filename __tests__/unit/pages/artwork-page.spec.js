import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueMeta from 'vue-meta'
import mockData from '@fixtures/mock-data.js'
import ArtworkPage from '@pages/_date.vue'

describe('Artwork Page', () => {
  it('should be a Vue instance', () => {
    const localVue = createLocalVue()
    localVue.use(VueMeta, { keyName: 'head' })

    const wrapper = shallowMount(ArtworkPage, {
      stubs: {
        'base-carousel': true,
      },
      mocks: {
        $store: {
          state: {
            activeId: '01-01',
            slides: [],
            currentArtworkHeight: 100,
          },
          getters: {
            getArtworkById() {
              return null
            },
          },
        },
        $route: {
          params: {
            date: '01-01',
          },
        },
      },
      localVue,
    })

    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    const localVue = createLocalVue()
    localVue.use(VueMeta, { keyName: 'head' })

    const wrapper = shallowMount(ArtworkPage, {
      stubs: {
        'base-carousel': true,
      },
      mocks: {
        $store: {
          state: {
            activeId: '01-01',
            slides: [],
            currentArtworkHeight: 100,
          },
          getters: {
            getArtworkById() {
              return mockData.artwork
            },
          },
        },
        $route: {
          params: {
            date: '01-01',
          },
        },
      },
      methods: {
        $icon() {
          return true
        },
      },
      localVue,
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('fetch will load store with slide data set initial active slide', async () => {
    const dispatchMock = jest.fn()

    const wrapper = shallowMount(ArtworkPage, {
      stubs: {
        'base-carousel': true,
      },
      mocks: {
        $store: {
          state: {
            activeId: '01-01',
            slides: [],
            currentArtworkHeight: 100,
          },
          getters: {
            getArtworkById() {
              return mockData.artwork
            },
          },
        },
        $route: {
          params: {
            date: '01-01',
          },
        },
      },
      methods: {
        $icon() {
          return true
        },
      },
    })

    await wrapper.vm.$options.fetch({
      store: {
        dispatch(...args) {
          dispatchMock(args)
        },
      },
      params: {
        date: '01-01',
      },
    })

    expect(dispatchMock.mock.calls[0][0][0]).toBe('updateSlideData')
    expect(dispatchMock.mock.calls[1][0][0]).toBe('updateCurrentSlideIndex')
    expect(dispatchMock.mock.calls[2][0][0]).toBe('updateCurrentPage')
  })

  it('should validate route params', () => {
    const wrapper = shallowMount(ArtworkPage, {
      stubs: {
        'base-carousel': true,
      },
      mocks: {
        $store: {
          state: {
            activeId: '01-01',
            slides: [],
            currentArtworkHeight: 100,
          },
          getters: {
            getArtworkById() {
              return null
            },
          },
        },
        $route: {
          params: {
            date: '01-01',
          },
        },
      },
    })

    const isValid = wrapper.vm.$options.validate({
      params: {
        date: '01-01',
      },
    })
    expect(isValid).toBe(true)

    const isNotValid = wrapper.vm.$options.validate({
      params: {
        date: '99-99',
      },
    })

    expect(isNotValid).toBe(false)
  })
})
