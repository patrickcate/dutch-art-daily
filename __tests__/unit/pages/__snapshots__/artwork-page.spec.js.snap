// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artwork Page renders correctly 1`] = `
<div>
  <main class="l-page__main">
    <base-carousel-stub name="carousel" keyboard="true" simulate-touch="true" class="l-page__carousel" style="height: 100px;"></base-carousel-stub>
  </main>
  <nav class="l-page__nav">
    <timeline-nav-button-stub direction="prev"></timeline-nav-button-stub>
    <timeline-nav-stub class="l-page__timeline"></timeline-nav-stub>
    <timeline-nav-button-stub direction="next"></timeline-nav-button-stub>
  </nav>
  <details-carousel-stub class="l-page__details"></details-carousel-stub>
</div>
`;
